tilings/algorithms/fusion.py

```diff
@@ -273,6 +292,39 @@ def new_assumption(self):
             or (not self._fuse_row and cell[0] == self._col_idx)
         )
 
+    def check_isolation_level(self):
+        """
+        Checks whether the requirements for self.isolation_level are met.
+        """
+        if self.isolation_level is None:
+            return True
+
+        if self._fuse_row:
+            fusing_cells = [
+                (i, self._row_idx) for i in range(self._tiling.dimensions[0])
+            ] + [(i, self._row_idx + 1) for i in range(self._tiling.dimensions[0])]
+        else:
+            fusing_cells = [
+                (self._col_idx, i) for i in range(self._tiling.dimensions[1])
+            ] + [(self._col_idx + 1, i) for i in range(self._tiling.dimensions[1])]
+        num_fusing_assumptions = len(
+            [
+                assumption
+                for assumption in self._tiling.assumptions
+                if any(cell in gp.pos for gp in assumption.gps for cell in fusing_cells)
+            ]
+        )
```

Comment:
I would create a method `_num_fusing_assumptions` then you will only make this if needed (i.e., in 'isolated' you first check if the number of assumptions are ok)

Reply:
Done.